crates/astria-conductor/src/conductor/mod.rs

```diff
@@ -0,0 +1,155 @@
+mod inner;
+
+use std::future::Future;
+
+use astria_eyre::eyre;
+use inner::{
+    ConductorInner,
+    InnerHandle,
+    RestartOrShutdown,
+};
+use pin_project_lite::pin_project;
+use tokio::task::{
+    JoinError,
+    JoinHandle,
+};
+use tokio_util::sync::CancellationToken;
+use tracing::{
+    info,
+    instrument,
+};
+
+use crate::{
+    metrics::Metrics,
+    Config,
+};
+
+pin_project! {
+    /// Handle to the conductor, returned by [`Conductor::spawn`].
+    pub struct Handle {
+        shutdown_token: CancellationToken,
+        task: Option<JoinHandle<eyre::Result<()>>>,
+    }
+}
+
+impl Handle {
+    /// Initiates shutdown of the conductor and returns its result.
+    ///
+    /// # Errors
+    /// Returns an error if the conductor exited with an error.
+    ///
+    /// # Panics
+    /// Panics if shutdown is called twice.
+    #[instrument(skip_all, err)]
+    pub async fn shutdown(&mut self) -> Result<eyre::Result<()>, JoinError> {
+        self.shutdown_token.cancel();
+        self.task
+            .take()
+            .expect("shutdown must not be called twice")
+            .await
+    }
+}
+
+impl Future for Handle {
+    type Output = Result<eyre::Result<()>, tokio::task::JoinError>;
+
+    fn poll(
+        self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Self::Output> {
+        use futures::future::FutureExt as _;
+        let this = self.project();
+        let task = this
+            .task
+            .as_mut()
+            .expect("the Conductor handle must not be polled after shutdown");
+        task.poll_unpin(cx)
+    }
+}
+
+/// A wrapper around [`ConductorInner`] that manages shutdown and restart of the conductor.
+pub struct Conductor {
+    /// Token to signal to all tasks to shut down gracefully.
+    shutdown_token: CancellationToken,
+
+    /// Handle for the inner conductor task.
+    handle: InnerHandle,
```

Comment:
"handle` is very nondescriptive.
```suggestion
    inner: inner::ConductorImpl
```